src/components/Banner.tsx

```diff
@@ -0,0 +1,71 @@
+import { useState } from "react";
+
+const Banner = () => {
+  let banner: HTMLTdsBannerElement;
+  const [bannerVisibility, setBannerVisibility] = useState(true);
+
+  const handleBanner = () => {
+    setBannerVisibility(!bannerVisibility);
+    bannerVisibility ? banner.hideBanner() : banner.showBanner();
+  };
```

Comment:
```suggestion
  const handleBanner = () => {
    if (banner) {
      setBannerVisibility(!bannerVisibility);
      bannerVisibility ? banner.hideBanner() : banner.showBanner();
    }
  };
```

Something like this would be needed as it trows "null" errors in the console, I guess as toggle and banner are in the same code so everything rerenders at the same time?